discussment-ws/docs/discussment-service-v1.yaml

```diff
@@ -0,0 +1,583 @@
+        required: true
+      description:
+        description: Description of the topic.
+        type: string
+        required: true
+    example: |
+      {
+        "name": "default topic",
+        "description": ""
+      }
+  TopicDTO:
+    type: object
+    properties:
+      id:
+        description: Unique ID of the topic.
+        type: integer
+        required: true
+      name:
+        description: Name of the topic.
+        type: string
+        required: true
+      description:
+        description: Description of the topic.
+        type: string
+        required: true
+    example: |
+      {
+        "id": 1,
+        "name": "default topic",
+        "description": ""
+      }
+  CreateDiscussionDTO:
+    type: object
+    properties:
+      name:
+        description: Name of the discussion.
+        type: string
+        required: false
+      password:
+        description: Password of the discussion in plain text. Pass empty string if the newly created discussion is not to be password protected.
+        type: string
+        required: false
+    example: |
+      {
+        "name": "default discussion",
+        "password": "password"
+      }
+  DiscussionDTO:
+    type: object
+    properties:
+      id:
+        description: Unique ID of the discussion.
+        type: integer
+        required: true
+      name:
+        description: Name of the discussion.
+        type: string
+        required: true
+      password:
+        description: Password of the discussion in plain text. Pass empty string if the newly created discussion is not to be password protected.
+        type: string
+        required: true
+    example: |
+      {
+        "id": 1,
+        "name": "default discussion",
+        "password": ""
+      }
+  CreatePostDTO:
+    type: object
+    properties:
+      text:
+        description: Text of the post.
+        type: string
+        required: true
+    example: |
+      {
+        "text": "Lorem ipsum."
+      }
+  UpdatePostDTO:
+    type: object
+    properties:
+      state:
+        description: State of the post.
+        type: PostState
+        required: true
+    example: |
+      {
+        "state": "DISABLED"
+      }
+  PostDTO:
+    description: Post in a discussion.
+    type: object
+    properties:
+      id:
+        description: Unique ID of the post.
+        type: integer
+        required: true
+      text:
+        description: Text of the post.
+        type: string
+        required: true
+      state:
+        description: State of the post.
+        type: PostState
+        required: true
+      level:
+        description: Nesting level of the post. Starts from 0, increases by 1 for every reply level.
+        type: integer
+        required: true
+      createdBy:
+        description: Username of the user who authored the post.
+        type: string
+        required: true
+      createdAt:
+        description: Datetime when the post was created in UTC timezone.
```

Comment:
Description should give explicit example of formatting. Datetimes are always trouble when de/serializing.

Reply:
Format is specified by RAML's `type: datetime-only`.

Reply:
When I read the documentation I dont know the format. Documentation needs to be explicit.